compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4258,26 +4303,69 @@ private String addReferencesAndReturnDocumentationText(LinkedList<BLangMarkdownR
                 BLangMarkdownReferenceDocumentation bLangRefDoc = new BLangMarkdownReferenceDocumentation();
                 DocumentationReferenceNode docReferenceNode = (DocumentationReferenceNode) element;
 
+                DiagnosticPos pos = getPosition(docReferenceNode);
+                bLangRefDoc.pos = pos;
+
+                Token startBacktick = docReferenceNode.startBacktick();
+                Node backtickContent = docReferenceNode.backtickContent();
+                Token endBacktick = docReferenceNode.endBacktick();
+
+                String contentString = backtickContent.isMissing() ? "" : backtickContent.toString();
+                bLangRefDoc.referenceName = contentString;
+
                 bLangRefDoc.type = DocumentationReferenceType.BACKTICK_CONTENT;
-                docReferenceNode.referenceType().ifPresent(
+                Optional<Token> referenceType = docReferenceNode.referenceType();
+                referenceType.ifPresent(
                         refType -> {
                             bLangRefDoc.type = stringToRefType(refType.text());
+                            if (backtickContent.kind() == SyntaxKind.BACKTICK_CONTENT) {
+                                // backtick content preceded by a special keyword, should be an identifier
+                                bLangRefDoc.hasParserWarnings = true;
+                                dlog.warning(pos, DiagnosticCode.INVALID_DOCUMENTATION_IDENTIFIER, contentString);
+                            } else {
+                                bLangRefDoc.identifier = ((Token) backtickContent).text();
+                            }
                             docText.append(refType.toString());
                         }
                 );
 
-                Token startBacktick = docReferenceNode.startBacktick();
-                docText.append(startBacktick.isMissing() ? "" : startBacktick.text());
+                if (!referenceType.isPresent() && backtickContent.kind() == SyntaxKind.BACKTICK_CONTENT) {
+                    // reaching here means, backtick content is not in one of x(), m:x(), T.y(), m:T.y() formats
+                    // no warning is logged for this case
+                    bLangRefDoc.hasParserWarnings = true;
+                }
 
-                Token backtickContent = docReferenceNode.backtickContent();
-                String contentString = backtickContent.isMissing() ? "" : backtickContent.text();
-                bLangRefDoc.referenceName = contentString;
-                docText.append(contentString);
+                if (!referenceType.isPresent() && backtickContent.kind() != SyntaxKind.BACKTICK_CONTENT) {
```

Comment:
can take `!referenceType.isPresent()` condition out, as its common for both conditions.

Also would be good to extract this content to a separate method?